spec/rubocop/cop/rails/enum_hash_spec.rb

```diff
@@ -1,162 +1,314 @@
 # frozen_string_literal: true
 
 RSpec.describe RuboCop::Cop::Rails::EnumHash, :config do
-  context 'when array syntax is used' do
-    context 'with %i[] syntax' do
-      it 'registers an offense' do
+  context 'when Rails 7 syntax is used' do
+    context 'when array syntax is used' do
+      context 'with %i[] syntax' do
+        it 'registers an offense' do
+          expect_offense(<<~RUBY)
+            enum :status, %i[active archived]
+                          ^^^^^^^^^^^^^^^^^^^ Enum defined as an array found in `status` enum declaration. Use hash syntax instead.
+          RUBY
```

Comment:
Can you follow up `expect_correction` with `expect_offense`? Please do this for the diffs in this PR.